OpenRA.Mods.D2k/Traits/Buildings/ProductionFromMapEdge.cs

```diff
@@ -24,59 +25,70 @@ class ProductionFromMapEdgeInfo : ProductionInfo
 
 	class ProductionFromMapEdge : Production
 	{
+		CPos closestEdgeCell;
+		WPos closestEdgeCellCenter;
+
 		public ProductionFromMapEdge(ActorInitializer init, ProductionInfo info)
 			: base(init, info) { }
 
-		public override bool Produce(Actor self, ActorInfo producee, string raceVariant)
+		public override bool Produce(Actor self, IEnumerable<ActorInfo> actorsToProduce, string raceVariant)
 		{
-			var location = self.World.Map.ChooseClosestEdgeCell(self.Location);
-			var pos = self.World.Map.CenterOfCell(location);
+			closestEdgeCell = self.World.Map.ChooseClosestEdgeCell(self.Location);
+			closestEdgeCellCenter = self.World.Map.CenterOfCell(closestEdgeCell);
+
+			foreach (var actorInfo in actorsToProduce)
+				DoProduction(self, actorInfo, null, raceVariant);
+
+			return true;
+		}
 
+		public override Actor DoProduction(Actor self, ActorInfo actorToProduce, ExitInfo exitInfo, string raceVariant)
+		{
 			// If aircraft, spawn at cruise altitude
-			var aircraftInfo = producee.Traits.GetOrDefault<AircraftInfo>();
+			var aircraftInfo = actorToProduce.Traits.GetOrDefault<AircraftInfo>();
 			if (aircraftInfo != null)
-				pos += new WVec(0, 0, aircraftInfo.CruiseAltitude.Range);
+				closestEdgeCellCenter += new WVec(0, 0, aircraftInfo.CruiseAltitude.Range);
 
-			var initialFacing = self.World.Map.FacingBetween(location, self.Location, 0);
+			var initialFacing = self.World.Map.FacingBetween(closestEdgeCell, self.Location, 0);
 
-			self.World.AddFrameEndTask(w =>
-				{
-					var td = new TypeDictionary
+			var td = new TypeDictionary
 					{
 						new OwnerInit(self.Owner),
-						new LocationInit(location),
-						new CenterPositionInit(pos),
+						new LocationInit(closestEdgeCell),
+						new CenterPositionInit(closestEdgeCellCenter),
 						new FacingInit(initialFacing)
 					};
 
-					if (raceVariant != null)
-						td.Add(new RaceInit(raceVariant));
+			if (raceVariant != null)
```

Comment:
Why not the same as [above](https://github.com/OpenRA/OpenRA/pull/7786/files#diff-3a1930006db09a5bddaed5ff0fb96974R69)?